File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/actor/AkkaMessageApi.java
package net.bestia.zoneserver.actor;

import akka.actor.ActorRef;
import net.bestia.messages.MessageApi;
import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class AkkaMessageApi implements MessageApi {

  private final static Logger LOG = LoggerFactory.getLogger(AkkaMessageApi.class);

  private ActorRef postmaster;

  @Override
  public void setPostmaster(@NotNull ActorRef postmaster) {
    this.postmaster = postmaster;
  }

  @Override
  public void send(@NotNull Object message) {
    LOG.debug("sendig: {}", message);
    postmaster.tell(message, ActorRef.noSender());
  }
}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/AkkaMessageApi.kt
package net.bestia.zoneserver

import akka.actor.ActorRef
import mu.KotlinLogging
import net.bestia.messages.MessageApi

private val LOG = KotlinLogging.logger { }

class AkkaMessageApi : MessageApi {

  private var postmaster: ActorRef? = null

  override fun setPostmaster(postmaster: ActorRef) {
    this.postmaster = postmaster
  }

  override fun send(message: Any) {
    LOG.debug { "Sendig: $message" }
    postmaster!!.tell(message, ActorRef.noSender())
  }
}
